.github/workflows/build.yml

```diff
@@ -141,3 +141,43 @@ jobs:
           dev_tags: |
             hashicorppreview/${{ env.REG_NAME }}:${{ env.version }}
             docker.io/hashicorppreview/${{ env.REG_NAME }}:${{ env.version }}
+
+  upload-dev-docker:
+    name: Upload dev image (hashicorpdev/consul-lambda-registrator:<commit>)
+    environment: dockerhub/hashicorpdev
+    needs:
+      - get-product-version
+      - build-docker-default
+    runs-on: ubuntu-latest
+    env:
+      repo: ${{ github.event.repository.name }}
+      version: ${{ needs.get-product-version.outputs.product-version }}
+      arch: amd64
+      target: release-default
+      git-short-sha: ${{ needs.get-product-version.outputs.git-short-sha }}
+    steps:
+      - name: Download image artifact
+        uses: actions/download-artifact@v3
+        with:
+          name: ${{ env.repo }}_${{ env.target }}_linux_${{ env.arch }}_${{ env.version }}_${{ github.sha }}.docker.dev.tar
+      - name: Docker push
+        shell: bash
+        run: |
+          TAG=hashicorpdev/${{ env.REG_NAME }}:${{ env.git-short-sha }}
+          echo "==> Load docker image from tar archive"
+          docker load -i "${{ env.repo }}_${{ env.target }}_linux_${{ env.arch }}_${{ env.version }}_${{ github.sha }}.docker.dev.tar"
+          echo "==> Tag docker image $TAG"
+          docker tag "hashicorppreview/${{ env.REG_NAME }}:${{ env.version }}" "$TAG"
+          echo "==> Docker login"
+          echo ${{ secrets.DOCKER_PASS }} | docker login -u=${{ secrets.DOCKER_USER }} --password-stdin
```

Comment:
Maybe a not that eventually we'd like to not rely on Github secrets (remembering https://github.com/hashicorp/consul-ecs/pull/90#discussion_r879271623)

Reply:
Thanks for the reminder Paul. I've added a ticket to do this in a future release so it doesn't fall through the cracks.